src/js/component.js

```diff
@@ -82,1182 +90,1103 @@ var Component = CoreObject.extend({
       this.enableTouchActivity();
     }
   }
-});
 
-/**
- * Dispose of the component and all child components
- */
-Component.prototype.dispose = function(){
-  this.trigger({ type: 'dispose', 'bubbles': false });
+  // Temp for ES6 class transition, remove before 5.0
+  init() {
+    // console.log('init called on Component');
+    Component.apply(this, arguments);
+  }
 
-  // Dispose all children.
-  if (this.children_) {
-    for (var i = this.children_.length - 1; i >= 0; i--) {
-      if (this.children_[i].dispose) {
-        this.children_[i].dispose();
+  /**
+   * Dispose of the component and all child components
+   */
+  dispose() {
+    this.trigger({ type: 'dispose', 'bubbles': false });
+
+    // Dispose all children.
+    if (this.children_) {
+      for (var i = this.children_.length - 1; i >= 0; i--) {
+        if (this.children_[i].dispose) {
+          this.children_[i].dispose();
+        }
       }
     }
-  }
 
-  // Delete child references
-  this.children_ = null;
-  this.childIndex_ = null;
-  this.childNameIndex_ = null;
+    // Delete child references
+    this.children_ = null;
+    this.childIndex_ = null;
+    this.childNameIndex_ = null;
 
-  // Remove all event listeners.
-  this.off();
-
-  // Remove element from DOM
-  if (this.el_.parentNode) {
-    this.el_.parentNode.removeChild(this.el_);
-  }
+    // Remove all event listeners.
+    this.off();
 
-  Lib.removeData(this.el_);
-  this.el_ = null;
-};
+    // Remove element from DOM
+    if (this.el_.parentNode) {
```

Comment:
The official Closure Library style is to use the private references everywhere internally so compiler can mangle them, and then export the methods for anything you want accessible externally. That's not always obvious for other contributors, and I also started to worry that people would start using the private properties externally after seeing internal code, but then on the other hand I think uglify has an option to mangle private props now if we wanted to keep that part of the file size savings.

Either way it's open for a change.